proversity_reports_script/report_backend/enrollment_per_site_report.py

```diff
@@ -160,13 +160,34 @@ def build_course_enrollment_per_site_report(enrollment_data):
         row_data['Username'] = enrollment.get('username', '')
         row_data['Email'] = enrollment.get('email', '')
         row_data['Role'] = enrollment.get('role', '')
-        row_data['Date of Enrollment'] = date_of_enrollment.strftime(date_format)
-        row_data['Date of Registration'] = date_of_registration.strftime(date_format)
-        row_data['Date of first access'] = date_of_registration.strftime(date_format)
-        row_data['Date of first access to course'] = date_of_first_access_to_course.strftime(date_format)
-        row_data['Date of Licence Expiration'] = (date_of_registration + one_year).strftime(date_format)
-        row_data['Days used'] = (datetime.now() - date_of_registration).days
-        row_data['Licence days remaining'] = 365 - (datetime.now() - date_of_registration).days
+        row_data['Date of Enrollment'] = (
+            date_of_enrollment.strftime(date_format)
+            if date_of_enrollment else ''
+        )
+        row_data['Date of Registration'] = (
+            date_of_registration.strftime(date_format)
+            if date_of_registration else ''
+        )
+        row_data['Date of first access'] = (
+            date_of_registration.strftime(date_format)
+            if date_of_registration else ''
+        )
+        row_data['Date of first access to course'] = (
+            date_of_first_access_to_course.strftime(date_format)
+            if date_of_first_access_to_course else ''
+        )
+        row_data['Date of Licence Expiration'] = (
+            (date_of_registration + one_year).strftime(date_format)
+            if date_of_registration else ''
+        )
+        row_data['Days used'] = (
+            (datetime.now() - date_of_registration).days
+            if date_of_registration else ''
+        )
+        row_data['Licence days remaining'] = (
+            365 - (datetime.now() - date_of_registration).days
```

Comment:
'365' looks like a magic number. Let's try to use a constant or setting if it makes sense @Squirrel18

Reply:
Addressed @diegomillan